operator/v1/types_console.go

```diff
@@ -146,6 +146,22 @@ type ConsoleCustomization struct {
 	// +listMapKey=id
 	// +optional
 	Perspectives []Perspective `json:"perspectives"`
+	// consoleURL is an URL to be used as the base console address.
+	// This field is intended for clusters without ingress capability,
+	// where access to routes is not possible.
+	// Invalid value will prevent a console rollout.
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self.startsWith('https://')",message="console url scheme must be https"
+	ConsoleURL string `json:"consoleURL,omitempty"`
+	// downloadsURL is an URL to be used as the address to download client binaries.
+	// This field is intended for clusters without ingress capability,
+	// where access to routes is not possible.
+	// Invalid value will prevent a console rollout.
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self.startsWith('https://')",message="downloads url scheme must be https"
+	DownloadsURL string `json:"downloadsURL,omitempty"`
```

Comment:
I am curious why download was made plural.  You are downloading a binary with a downloadURL or a downloadsURL?

Reply:
There are multiple binary artifacts which are exposed on this address. So historically the route and the corresponding DNS name were plural. [Jakub asked me](https://github.com/openshift/api/pull/1887#discussion_r1600123520) to keep the naming consistent.
cc @jhadvig.

Reply:
What is this expecting to download, is it OC? You description says binaries so I guess it must be more than that?

Reply:
Yes, it's `oc` binary for different architectures.

Reply:
Considered calling it `clientDownloadURL`? Might be a bit more obvious if it had client in the name

Reply:
@jhadvig: What do you think about `clientDownloadURL`? Or maybe in the plural form: `clientDownloadsURL`?

Reply:
@alebedev87 I would stick to `clientDownloadsURL`.
Also thinking if we could nest it:
```go
clients:
  consoleURL: https://url-for-console.com
  downloadsURL: https://url-for-downloads.com
```
although I know that we are under time pressure 

@JoelSpeed @hexfusion just for clarification, all the downloads related resources(service, route, pod, deployment) use the plural name.

Reply:
Why does the consoleURL come under this clients categorisation? I'm not sure I'm following what these are actually for yet

Reply:
@JoelSpeed just a though on how to make both URLs nested under one field.

Reply:
> I'm not sure I'm following what these are actually for yet

The console and downloads workloads are exposed via `Route` objects of the same names. The OpenShift router resides at the edge of the cluster to handle all the traffic for all the `Route` objects. The ingress capability has been introduced in OCP 4.16. The ingress capability includes the OpenShift router. Therefore on the clusters where the ingress capability can be disabled (HyperShift ones) the console and downloads routes are left unserved.
The console operator which manages the console and downloads workloads relies on the presence of the routes: to do some health check pings and to manage authentication flows. With this new fields we enable user to provide an alternative (to route host) ingress URL. We can imagine a user which disabled the ingress capability on a ROSA cluster and used ALBO to provision AWS ALBs for its console and downloads deployments. The hostnames of these ALBs can be provided to the console operator to keep doing its duty: health checks, authentication management.

Reply:
> Therefore on the clusters where the ingress capability can be disabled (HyperShift ones) the console and downloads routes are left unserved.

What prevents me disabling the capability on a regular OCP cluster?

Ok, that context is helpful. Makes me thing this is something like `clientEndpoints` or `routeEndpoints`?

What would happen with these fields if they were set by the user and the Route objects existed as normal, does this take precedence?

Reply:
> just a though on how to make both URLs nested under one field

I can think of this:
```yaml
spec:
  customization:
     ingress:
       consoleURL: https://console-alb.com
       clientDownloadsURL: https://downloads-alb.com
```

Reply:
I like that suggestion, happy to go with that

Reply:
> What prevents me disabling the capability on a regular OCP cluster?

[The CVO doesn't allow this](https://github.com/openshift/cluster-version-operator/pull/946).

> Makes me thing this is something like clientEndpoints or routeEndpoints?

Yes. Not sure whether `endpoint` is the right word though. I would say it's like `ingress` status in `Service` object.

> What would happen with these fields if they were set by the user and the Route objects existed as normal, does this take precedence?

No. The fields will only be used when the ingress is disabled (== routes are not admitted but they may exist).

Reply:
Implemented the grouping of the new fields under `ingress` as described [here](https://github.com/openshift/api/pull/1887#discussion_r1601522967).